test/conftest.py

```diff
@@ -66,3 +99,83 @@ def pytest_collection_modifyitems(config, items):
             if run_id:
                 item._nodeid = f'{item._nodeid}.{run_id}'
                 item.name = f'{item.name}.{run_id}'
+
+
+def prepare_dir(dirname: str, pattern: str) -> None:
+    # Ensure the dir exists
+    pathlib.Path(dirname).mkdir(parents=True, exist_ok=True)
+    # Remove old artifacts
+    for p in glob.glob(os.path.join(dirname, pattern), recursive=True):
+        pathlib.Path(p).unlink()
+
+
+def prepare_dirs(tempdir_base: str, modes: list[str]) -> None:
+    prepare_dir(tempdir_base, "*.log")
+    for mode in modes:
+        prepare_dir(os.path.join(tempdir_base, mode), "*.log")
+        prepare_dir(os.path.join(tempdir_base, mode), "*.reject")
+        prepare_dir(os.path.join(tempdir_base, mode, "xml"), "*.xml")
+        shutil.rmtree(os.path.join(tempdir_base, mode, "failed_test"), ignore_errors=True)
+        prepare_dir(os.path.join(tempdir_base, mode, "failed_test"), "*")
+        prepare_dir(os.path.join(tempdir_base, mode, "allure"), "*.xml")
+        if TEST_RUNNER == "pytest":
+            shutil.rmtree(os.path.join(tempdir_base, mode, "pytest"), ignore_errors=True)
+            prepare_dir(os.path.join(tempdir_base, mode, "pytest"), "*")
+
+
+@universalasync.async_to_sync_wraps
+async def start_s3_mock_services(minio_tempdir_base: str) -> None:
+    ms = MinioServer(
+        tempdir_base=minio_tempdir_base,
+        address="127.0.0.1",
+        logger=LogPrefixAdapter(logger=logging.getLogger("minio"), extra={"prefix": "minio"}),
+    )
+    await ms.start()
+    TestSuite.artifacts.add_exit_artifact(None, ms.stop)
+
+    hosts = HostRegistry()
+    TestSuite.artifacts.add_exit_artifact(None, hosts.cleanup)
+
+    mock_s3_server = MockS3Server(
+        host=await hosts.lease_host(),
+        port=2012,
+        logger=LogPrefixAdapter(logger=logging.getLogger("s3_mock"), extra={"prefix": "s3_mock"}),
+    )
+    await mock_s3_server.start()
+    TestSuite.artifacts.add_exit_artifact(None, mock_s3_server.stop)
+
+    minio_uri = f"http://{os.environ[ms.ENV_ADDRESS]}:{os.environ[ms.ENV_PORT]}"
+    proxy_s3_server = S3ProxyServer(
+        host=await hosts.lease_host(),
+        port=9002,
+        minio_uri=minio_uri,
+        max_retries=3,
+        seed=int(time.time()),
+        logger=LogPrefixAdapter(logger=logging.getLogger("s3_proxy"), extra={"prefix": "s3_proxy"}),
+    )
+    await proxy_s3_server.start()
+    TestSuite.artifacts.add_exit_artifact(None, proxy_s3_server.stop)
+
+
+def pytest_sessionstart(session: pytest.Session) -> None:
+    # test.py starts S3 mock and create/cleanup testlog by itself. Also, if we run with --collect-only option,
+    # we don't need this stuff.
+    if TEST_RUNNER != "pytest" or session.config.getoption("--collect-only"):
```

Comment:
Nitpick: the TEST_RUNNER check (I don't know what is TEST_RUNNER, but I can guess) isn't needed if test.py doesn't pass "--test-py-init" it won't happen anyway!

Reply:
I prefer to keep these things separate. Anyway, both are temporary workarounds for the transition time.